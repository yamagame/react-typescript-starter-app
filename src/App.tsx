import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { useAppDispatch } from 'app/hooks';
import { fetchPageAsync } from 'features/article/articleSlice';
import { CounterPage } from 'pages/CounterPage';
import { TopPage } from 'pages/TopPage';
import { ArticlePage } from 'pages/ArticlePage';

export function App() {
  const dispatch = useAppDispatch();

  React.useEffect(() => {
    dispatch(fetchPageAsync());
  }, [dispatch]);

  return (
    <BrowserRouter>
      <Routes>
        {/*-------------- トップページ */}
        <Route path="/" element={<TopPage />} />
        {/*-------------- トップページ */}
        <Route path="/counter" element={<CounterPage />} />
        {/*-------------- 各ページ */}
        <Route path={'article/:pageId'} element={<ArticlePage />} />
      </Routes>
    </BrowserRouter>
  );
}
