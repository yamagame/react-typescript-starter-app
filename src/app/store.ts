import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import articleReducer from '../features/article/articleSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    article: articleReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
