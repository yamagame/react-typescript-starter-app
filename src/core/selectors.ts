import { applicationSelector } from 'features/application/selector';
import { articleSelector } from 'features/article/selector';
import { counterSelector } from 'features/counter/selector';
import { formSampleSelector } from 'features/formSample/selector';
import { todoSelector } from 'features/todo/selector';
import { todoCounterSelector } from 'features/todoCounter/selector';

// ----------------------------------------------- selectors
export const selectors = {
  application: applicationSelector,
  article: articleSelector,
  counter: counterSelector,
  formSample: formSampleSelector,
  todo: todoSelector,
  todoCounter: todoCounterSelector,
};
