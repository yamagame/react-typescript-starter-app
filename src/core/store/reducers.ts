import counterReducer from 'core/slices/counter';
import articleReducer from 'core/slices/article';
import formSampleReducer from 'core/slices/formSample';
import todoReducer from 'core/slices/todo';

export const reducer = {
  counter: counterReducer,
  article: articleReducer,
  formSample: formSampleReducer,
  todo: todoReducer,
};
