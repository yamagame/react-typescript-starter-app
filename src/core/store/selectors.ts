import { RootState } from '.';

export const selectors = {
  article: (state: RootState) => state.article,
  counter: (state: RootState) => state.counter,
  formSample: (state: RootState) => state.formSample,
  todo: (state: RootState) => state.todo,
};
