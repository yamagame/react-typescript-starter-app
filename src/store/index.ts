import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

import applicatoinReducer from 'features/application';
import counterReducer from 'features/counter';
import articleReducer from 'features/article';
import formSampleReducer from 'features/formSample';
import todoReducer from 'features/todo';

export const store = configureStore({
  reducer: {
    application: applicatoinReducer,
    counter: counterReducer,
    article: articleReducer,
    formSample: formSampleReducer,
    todo: todoReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const selectors = {
  application: (state: RootState) => state.application,
  article: (state: RootState) => state.article,
  counter: (state: RootState) => state.counter,
  formSample: (state: RootState) => state.formSample,
  todo: (state: RootState) => state.todo,
};
