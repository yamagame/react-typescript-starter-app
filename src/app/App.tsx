import React from 'react';
import { useAppDispatch } from 'store';
import { actions } from 'features/article';
import { AppRouter } from 'routers/ApplicationRouter';

export function App() {
  const dispatch = useAppDispatch();

  React.useEffect(() => {
    dispatch(actions.fetchPagesAsync());
  }, [dispatch]);

  return <AppRouter />;
}
