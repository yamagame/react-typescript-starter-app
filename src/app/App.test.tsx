import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from 'core/store';
import { App } from 'app/App';

test('renders learn react link', () => {
  const { container } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );

  expect(container).toMatchSnapshot();
});
