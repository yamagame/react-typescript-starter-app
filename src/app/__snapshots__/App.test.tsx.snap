// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appの表示 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      ListPage
    </h1>
    <p>
      1.0.0
    </p>
    <div>
      <a
        href="/counter"
      >
        Counter
      </a>
    </div>
    <div>
      <a
        href="/form"
      >
        Form
      </a>
    </div>
    <div>
      <a
        href="/form-formik"
      >
        Form-Formik
      </a>
    </div>
    <div>
      <a
        href="/my-page"
      >
        MyPage
      </a>
    </div>
    <div>
      <a
        href="/test-page"
      >
        TestPage
      </a>
    </div>
    <div>
      <a
        href="/todo"
      >
        ToDoPage
      </a>
    </div>
    <div>
      <a
        href="/todo-counter"
      >
        ToDoCounter
      </a>
    </div>
    <div>
      <a
        href="/d3sample"
      >
        D3Sample
      </a>
    </div>
  </div>
</div>
`;
